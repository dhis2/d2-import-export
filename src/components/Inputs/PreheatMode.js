import i18n from '@dhis2/d2-i18n'
import React from 'react'
import { RadioGroup } from '../FinalFormComponents/RadioGroup'
import { Field } from '../Field/Field'
import { Label } from '../Field/Label'

export const OPTION_REFERENCE = {
    value: 'REFERENCE',
    label: i18n.t('Reference'),
}
export const OPTION_ALL = { value: 'ALL', label: i18n.t('All') }
export const OPTION_NONE = { value: 'NONE', label: i18n.t('None') }

export const PREHEAT_MODE_KEY = 'preheatMode'
export const PREHEAT_MODE_DEFAULT_VALUE = OPTION_REFERENCE.value

export const PreheatMode = () => (
    <Field>
        <Label>{i18n.t('Preheat mode')}</Label>
        <RadioGroup
            name={PREHEAT_MODE_KEY}
            label={i18n.t('Preheat mode')}
            options={[OPTION_REFERENCE, OPTION_ALL, OPTION_NONE]}
            dataTest="input-preheat-mode"
        />
    </Field>
)
