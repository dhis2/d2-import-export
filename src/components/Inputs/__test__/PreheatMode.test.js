import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { PreheatMode } from '../PreheatMode'

jest.mock('react-final-form')

describe('Input component - PreheatMode', () => {
    it('should render correctly', () => {
        const file = shallow(<PreheatMode />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
