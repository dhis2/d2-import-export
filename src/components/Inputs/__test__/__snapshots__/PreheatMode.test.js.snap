// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - PreheatMode should render correctly 1`] = `
<Field>
  <Label>
    Preheat mode
  </Label>
  <RadioGroup
    dataTest="input-preheat-mode"
    label="Preheat mode"
    name="preheatMode"
    options={
      Array [
        Object {
          "label": "Reference",
          "value": "REFERENCE",
        },
        Object {
          "label": "All",
          "value": "ALL",
        },
        Object {
          "label": "None",
          "value": "NONE",
        },
      ]
    }
  />
</Field>
`;
